tests/unit-tests/src/VuFindTest/OaiPmh/HarvesterTest.php

```diff
@@ -261,50 +261,35 @@ protected function getFakeResponseWithToken()
 XML;
     }
 
-    /**
-     * Test that granularity is autoloaded by default.
-     *
-     * @return void
-     */
-    public function testGranularityAutoload()
-    {
-        $comm = $this->getMockCommunicator();
-        $comm->expects($this->once(0))->method('request')
-            ->with($this->equalTo('Identify'))
-            ->will(
-                $this->returnValue(
-                    simplexml_load_string($this->getFakeIdentifyResponse())
-                )
-            );
-        $harvester = $this->getHarvester([], $comm);
-        $this->assertEquals(
-            'YYYY-MM-DDThh:mm:ssZ',
-            $this->getProperty($harvester, 'granularity')
-        );
-    }
-
     /**
      * Test that a single ListRecords call with no resumption token triggers
-     * a write to the writer.
+     * a write to the writer and persists a harvest end date that's based on
+     * the OAI host response date.
      *
      * @return void
      */
     public function testSimpleListRecords()
     {
         $comm = $this->getMockCommunicator();
         $expectedSettings = ['metadataPrefix' => 'oai_dc'];
-        $comm->expects($this->once())->method('request')
-            ->with($this->equalTo('ListRecords'), $this->equalTo($expectedSettings))
-            ->will($this->returnValue(simplexml_load_string($this->getFakeResponse())));
+        $comm->expects($this->exactly(2))->method('request')
+            ->withConsecutive(
+                ['Identify', []],
+                ['ListRecords', $expectedSettings],
+            )
+            ->willReturnOnConsecutiveCalls(
+                simplexml_load_string($this->getFakeIdentifyResponse()),
+                simplexml_load_string($this->getFakeResponse())
+            );
         $writer = $this->getMockRecordWriter();
         $writer->expects($this->once())->method('write')
             ->with($this->isInstanceOf('SimpleXMLElement'))
             ->will($this->returnValue(1468434382));
         $sm = $this->getMockStateManager();
         $sm->expects($this->once())->method('saveDate')
-            ->with($this->equalTo('2016-07-13T14:26:22Z'));
+            ->with($this->equalTo('2016-07-12T16:19:54Z'));
```

Comment:
This date now comes directly from the mocked Identify responseDate. So asserting a match to that responseDate here should provide coverage for the new end date handling that's central to #7. See also the new testSimpleListRecordsGranularityHandling().